include/revng/Model/Types.h

```diff
@@ -0,0 +1,335 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <algorithm>
+#include <optional>
+
+#include "llvm/ADT/StringRef.h"
+
+#include "revng/ADT/SortedVector.h"
+#include "revng/Support/Assert.h"
+#include "revng/Support/Debug.h"
+
+namespace flatc {
```

Comment:
Is there a real benefit in having a namespace here?
`flatc` is just a part of the model (`model` namespace).
I was thinking about `flatc` more as an informal thing than something widespread in the codebase.

Reply:
The point is, we already have `llvm::Type`, and `clang::Type` in `revng-c`. Given that `revng-c` is going to be one of the heaviest users of this type system, I think we need a way to distinguish `flatc::Type` from others. But I can do with `model::Type` if you like it better.

Reply:
Yeah, `flatc` is just `model` after all. Go for `model::Type`.

Reply:
Done.